skia-bindings/src/bindings.cpp

```diff
@@ -2644,6 +2644,99 @@ bool C_SkRuntimeEffect_allowBlender(const SkRuntimeEffect* self) {
     return self->allowBlender();
 }
 
+void C_SkRuntimeShaderBuilder_Construct(SkRuntimeShaderBuilder *uninitialized,
+                                        SkRuntimeEffect *effect) {
+  new (uninitialized) SkRuntimeShaderBuilder(sp(effect));
+}
+
+void C_SkRuntimeShaderBuilder_destruct(SkRuntimeShaderBuilder *self) {
+  self->~SkRuntimeShaderBuilder();
+}
+
+const SkRuntimeEffect *
+C_SkRuntimeShaderBuilder_effect(const SkRuntimeShaderBuilder *self) {
+  return self->effect();
+}
+
+sk_sp<const SkData>
+C_SkRuntimeShaderBuilder_uniforms(const SkRuntimeShaderBuilder *self) {
+  return self->uniforms();
+}
+
+SkSpan<const SkRuntimeEffect::ChildPtr>
+C_SkRuntimeShaderBuilder_children(const SkRuntimeShaderBuilder *self) {
+  return self->children();
+}
+
+SkShader *
+C_SkRuntimeShaderBuilder_makeShader(const SkRuntimeShaderBuilder *self,
+                                    const SkMatrix *localMatrix) {
+  auto shader = self->makeShader(localMatrix);
+  return shader.release();
+}
+
+void C_SkRuntimeShaderBuilder_setUniformFloat(SkRuntimeShaderBuilder *self,
+                                              const char *name, size_t count,
+                                              const float *const values,
+                                              size_t len) {
+  using float2 = std::array<float, 2>;
+  using float3 = std::array<float, 3>;
+  using float4 = std::array<float, 4>;
+  using float3x3 = std::array<float, 9>;
+  using float4x4 = std::array<float, 16>;
+
+  switch (len) {
+  case 1:
+    self->uniform(std::string_view(name, count)) = *values;
+    break;
+  case 2:
+    self->uniform(std::string_view(name, count)) =
+        *reinterpret_cast<const float2 *>(values);
+    break;
+  case 3:
+    self->uniform(std::string_view(name, count)) =
+        *reinterpret_cast<const float3 *>(values);
+    break;
+  case 4:
+    self->uniform(std::string_view(name, count)) =
+        *reinterpret_cast<const float4 *>(values);
+    break;
+  case 9:
+    self->uniform(std::string_view(name, count)) =
+        *reinterpret_cast<const float3x3 *>(values);
+    break;
+  case 16:
+    self->uniform(std::string_view(name, count)) =
+        *reinterpret_cast<const float4x4 *>(values);
+    break;
+  }
+}
+
+void C_SkRuntimeShaderBuilder_setUniformInt(SkRuntimeShaderBuilder *self,
+                                            const char *name, size_t count,
+                                            const int *const values,
+                                            size_t len) {
+  using int2 = std::array<int, 2>;
+  using int3 = std::array<int, 3>;
+  using int4 = std::array<int, 4>;
+  switch (len) {
```

Comment:
For this and the other method, if the provided slice of ints/floats in Rust doesn't match the expected length, the function will just quietly do nothing. That may make it difficult to debug.

I suggest to document the supported slice sizes in the Rust API docs of `set_uniform_float` and `set_uniform_int`, and perhaps even return an error if no size matched.

Reply:
makes sense, will include the suggestions

Reply:
@tronical  the PR is updated, the rust method is documented and now return a Result

Reply:
Love it, the custom error version looks great